File: floor_manage_backend/src/models/FloorPlan.js
const { DataTypes } = require('sequelize');
const sequelize = require('../config/database');

const FloorPlan = sequelize.define('FloorPlan', {
    id: {
        type: DataTypes.INTEGER,
        autoIncrement: true,
        primaryKey: true,
    },
    version: {
        type: DataTypes.INTEGER,
        allowNull: false,
    },
    image_url: {
        type: DataTypes.STRING,
        allowNull: false,
    },
    data_json: {
        type: DataTypes.TEXT,
        allowNull: true,
    },
    created_by: {
        type: DataTypes.INTEGER,
        allowNull: false,
    },
}, {
    indexes: [
        {
            unique: true,
            fields: ['version'],
        }
    ]
});

module.exports = FloorPlan;

File: floor_manage_ui/lib/features/admin/models/floor_plan.dart
class FloorPlan {
  final String id;
  final String name;
  final String imageUrl;
  final DateTime uploadedAt;
  final int version;

  FloorPlan({
    required this.id,
    required this.name,
    required this.imageUrl,
    required this.uploadedAt,
    required this.version,
  });
}
